version: '3.9'

services:
  db:
    image: postgres:14-alpine
    restart: always
    env_file:
      - env.yaml
    networks:
      - default
    ports:
      - 5432:5432
    volumes:
      - ./database.sql:/docker-entrypoint-initdb.d/solomon.sql
      - fipe_data:/var/lib/postgresql/data
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U postgres" ]
      interval: 2s
      timeout: 5s
      retries: 10

  pipe:
    build:
      context: ./src
      dockerfile: Dockerfile
    env_file:
      - env.yaml
    environment:
      - PYTHONUNBUFFERED=1
    depends_on:
      db:
        condition: service_healthy
    volumes:
      - ./src:/app
    networks:
      - default
    entrypoint: ["python", "main.py"]

networks:
  default:
    driver: bridge

volumes:
  fipe_data:
    external: true